cmd/repo-updater/repoupdater/server.go

```diff
@@ -432,7 +432,11 @@ func (s *Server) handleExternalServiceNamespaces(w http.ResponseWriter, r *http.
 
 	if err = genericSrc.CheckConnection(ctx); err != nil {
 		result = &protocol.ExternalServiceNamespacesResult{Error: err.Error()}
-		s.respond(w, http.StatusServiceUnavailable, result)
+		if errcode.IsUnauthorized(err) {
+			s.respond(w, http.StatusUnauthorized, result)
+		} else {
+			s.respond(w, http.StatusServiceUnavailable, result)
+		}
 		return
```

Comment:
Nit: You could use an early-return to reduce the if-else nesting. 

```suggestion
		if errcode.IsUnauthorized(err) {
			s.respond(w, http.StatusUnauthorized, result)
			return
		}

		s.respond(w, http.StatusServiceUnavailable, result)
		return
```